text/0756-helper-default-manager.md

```diff
@@ -0,0 +1,276 @@
+---
+Stage: Accepted
+Start Date: 2021-05-17
+Release Date: Unreleased
+Release Versions:
+  ember-source: vX.Y.Z
+  ember-data: vX.Y.Z
+Relevant Team(s): Ember.js, Learning
+RFC PR: https://github.com/emberjs/rfcs/pull/756
+---
+
+# Default Helper Manager
+
+## Summary
+
+Anything that can be in a template has its lifecycle managed by the manager pattern.
+Today, we have known managers for `@glimmer/component`, `@ember/helper`, etc.
+But what happens when the VM encounters an object for which there is no manager?,
+such as a plain function? This RFC proposes a default behavior for
+those unknown scenarios when it comes to _helpers_.
+
+## Motivation
+
+The addon, [ember-could-get-used-to-this](https://github.com/pzuraq/ember-could-get-used-to-this)
+demonstrated that it's possible to use plain functions for helpers and modifiers.
+And since Ember 3.25, helpers can be invoked directly from value references, this opened
+a whole new world of ergonomics improvements where a dev could define a function in a
+component class and use that function **as** a helper, thanks to
+ember-could-get-used-to-this implementing a Helper Manager that _knew what to do with plain functions.
+
+This has the impact of greatly reducing helper burden for apps and addon devs, in that,
+folks no longer need to jump over to the app/addon helpers directory to create a helper
+"just to do this one simple thing". It's all now `{{this.myHelper}}` or `{{this.myModifier}}`.
+
+This has the added benefit of, with template strict mode, using plain functions in module-scope
+for helpers and modifiers.
+
+For Example:
+
+<!--
+  If you're reading this RFC non-rendered, ignore the "jsx" language tag.
+  "jsx" is incorrect, and isn't perfect, but "gjs" syntax highlighting
+  hasn't been added to GitHub's highlighting yet.
+-->
+
+```jsx
+const double = num => num * 2;
+const resizable = element => { /* ... */ };
+
+<template>
+  {{double 2}}
+  <div {{resizable}}></div>
+</template>
+```
+_Note: `<template>` is experimental syntax, and should not be be the focus of this example_
+
+Another aspect of which is exciting is that it becomes easier, in tests to grab
+the output of yielding components:
+
+```jsx
+test('...', async function (assert) {
+  let output = [];
+  const capture = (...args) => output = args;
+
+  await render(hbs`
+    <MyComponent as |yielded info and things|>
+       {{capture yielded info and things}}
+    </MyComponent>
+  `);
+
+  assert.equal(output[0], /* value of yielded */ );
+  assert.equal(output[1], /* value of info */ );
+  // ...
+})
+```
+
+A default modifier manager will be covered in a different RFC.
+
+## Detailed design
+
+The logic for choosing when to use a default manager is, at a high-level:
+
+```
+if (noExistingManagerFor(X)) {
+  return defaultManager;
+}
```

Comment:
This needs to be fleshed out a bit more. Can you update to refer to the details laid out in the helper manager RFC?